libethereum/EthereumHost.cpp

```diff
@@ -512,26 +514,30 @@ tuple<vector<shared_ptr<EthereumPeer>>, vector<shared_ptr<EthereumPeer>>, vector
     vector<shared_ptr<EthereumPeer>> chosen;
     vector<shared_ptr<EthereumPeer>> allowed;
     vector<shared_ptr<SessionFace>> sessions;
-
-    size_t peerCount = 0;
+    double percentDecimal = _percent / 100.0;
     foreachPeer([&](std::shared_ptr<EthereumPeer> _p)
     {
         if (_allow(_p.get()))
         {
-            allowed.push_back(_p);
             sessions.push_back(_p->session());
+            allowed.push_back(_p);
         }
-        ++peerCount;
         return true;
     });
 
-    size_t chosenSize = (peerCount * _percent + 99) / 100;
+    if(_percent == 0 || allowed.size() == 0)
```

Comment:
There should be whitespace after `if`
Better use clang-format as described in https://github.com/ethereum/aleth/blob/master/CONTRIBUTING.md